### ISSUE TITLE
Tutorial 1st step fails in Jupyter

### ISSUE THREAD

**Issue by hannesbecher** [elissasoroj/shadie#17]
Hi,

I've managed to set up shadie (I think) as I have described in issue #16 

Unfortunately, I fail to run the first step of the [tutorial](https://elissasoroj.github.io/shadie/notebooks/shadie-v0_1_1-demo/).

![image](https://github.com/elissasoroj/shadie/assets/10515056/c595fa36-9228-40b9-b2dc-49e2b2cc2c6b)

I wonder if this may be a version problem with the dependency toytree? If so, you may want to specify versions for your dependency packages, so anaconda knows which ones to download.
```
# To look at my installed packages, I did:
conda activate shadie
conda list
```
This gives me:
```
# packages in environment at /Users/hbecher/miniconda3/envs/shadie:
#
# Name                    Version                   Build  Channel
altair                    5.2.0              pyhd8ed1ab_0    conda-forge
appnope                   0.1.3           py312hca03da5_1001  
arrow                     1.3.0              pyhd8ed1ab_0    conda-forge
asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge
attrs                     23.1.0             pyh71513ae_1    conda-forge
brotli-python             1.1.0           py312h9f69965_1    conda-forge
bzip2                     1.0.8                h93a5062_5    conda-forge
ca-certificates           2023.11.17           hf0a4a13_0    conda-forge
cairo                     1.18.0               hd1e100b_0    conda-forge
certifi                   2023.11.17      py312hca03da5_0  
chardet                   5.2.0           py312h81bd7bf_1    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
comm                      0.1.2           py312hca03da5_0  
custom-inherit            2.4.1              pyhd8ed1ab_0    conda-forge
debugpy                   1.6.7           py312h313beb8_0  
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
demes                     0.2.3              pyhd8ed1ab_0    conda-forge
exceptiongroup            1.2.0              pyhd8ed1ab_0    conda-forge
executing                 2.0.1              pyhd8ed1ab_0    conda-forge
expat                     2.5.0                hb7217d7_1    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_1    conda-forge
fontconfig                2.14.2               h82840c6_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
freetype                  2.12.1               hadb7bae_2    conda-forge
freetype-py               2.3.0              pyhd8ed1ab_0    conda-forge
future                    0.18.3             pyhd8ed1ab_0    conda-forge
gettext                   0.21.1               h0186832_0    conda-forge
gsl                       2.7                  h6e638da_0    conda-forge
humanize                  4.9.0                    pypi_0    pypi
icu                       73.2                 hc8870d7_0    conda-forge
idna                      3.6                pyhd8ed1ab_0    conda-forge
importlib-metadata        7.0.0              pyha770c72_0    conda-forge
importlib_resources       6.1.1              pyhd8ed1ab_0    conda-forge
ipykernel                 6.26.0             pyh3cd1d5f_0    conda-forge
ipython                   8.18.1             pyh707e725_3    conda-forge
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
jsonschema                4.20.0             pyhd8ed1ab_0    conda-forge
jsonschema-specifications 2023.11.2          pyhd8ed1ab_0    conda-forge
jupyter_client            8.6.0           py312hca03da5_0  
jupyter_core              5.5.0           py312hca03da5_0  
kastore                   0.3.2                    pypi_0    pypi
lcms2                     2.16                 ha0e7c42_0    conda-forge
lerc                      4.0.0                h9a09cb3_0    conda-forge
libblas                   3.9.0           20_osxarm64_openblas    conda-forge
libcblas                  3.9.0           20_osxarm64_openblas    conda-forge
libcxx                    16.0.6               h4653b0c_0    conda-forge
libdeflate                1.19                 hb547adb_0    conda-forge
libexpat                  2.5.0                hb7217d7_1    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libgfortran               5.0.0           13_2_0_hd922786_1    conda-forge
libgfortran5              13.2.0               hf226fd6_1    conda-forge
libglib                   2.78.3               hb438215_0    conda-forge
libiconv                  1.17                 h0d3ecfb_1    conda-forge
libjpeg-turbo             3.0.0                hb547adb_1    conda-forge
liblapack                 3.9.0           20_osxarm64_openblas    conda-forge
libopenblas               0.3.25          openmp_h6c19121_0    conda-forge
libpng                    1.6.39               h76d750c_0    conda-forge
libsodium                 1.0.18               h1a28f6b_0  
libsqlite                 3.44.2               h091b4b1_0    conda-forge
libtiff                   4.6.0                ha8a6c65_2    conda-forge
libwebp-base              1.3.2                hb547adb_0    conda-forge
libxcb                    1.15                 hf346824_0    conda-forge
libzlib                   1.2.13               h53f4e23_5    conda-forge
llvm-openmp               17.0.6               hcd81f8e_0    conda-forge
loguru                    0.7.2           py312h81bd7bf_1    conda-forge
markupsafe                2.1.3           py312h02f2b3b_1    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
msprime                   1.3.0           py312h0767fef_0    conda-forge
multipledispatch          0.6.0                      py_0    conda-forge
ncurses                   6.4                  h463b476_2    conda-forge
nest-asyncio              1.5.6           py312hca03da5_0  
newick                    1.9.0                    pypi_0    pypi
numpy                     1.26.2          py312h5d55045_0    conda-forge
openjpeg                  2.5.0                h4c1507b_3    conda-forge
openssl                   3.2.0                h0d3ecfb_1    conda-forge
packaging                 23.2               pyhd8ed1ab_0    conda-forge
pandas                    2.1.4           py312h9e53831_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pcre2                     10.42                h26f9a81_0    conda-forge
pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    10.1.0          py312hac22aec_0    conda-forge
pip                       23.3.1             pyhd8ed1ab_0    conda-forge
pixman                    0.42.2               h13dd4ca_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_1    conda-forge
platformdirs              3.10.0          py312hca03da5_0  
prompt-toolkit            3.0.42             pyha770c72_0    conda-forge
psutil                    5.9.0           py312h80987f9_0  
pthread-stubs             0.4               h27ca646_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pycairo                   1.25.1          py312h6cb0442_0    conda-forge
pygments                  2.17.2             pyhd8ed1ab_0    conda-forge
pyparsing                 3.1.1              pyhd8ed1ab_0    conda-forge
pypng                     0.20220715.0       pyhd8ed1ab_0    conda-forge
pyslim                    1.0.4                    pypi_0    pypi
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.12.0          h47c9636_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-tzdata             2023.3             pyhd8ed1ab_0    conda-forge
python_abi                3.12                    4_cp312    conda-forge
pytz                      2023.3.post1       pyhd8ed1ab_0    conda-forge
pyzmq                     25.1.0          py312h313beb8_0  
readline                  8.2                  h92ec313_1    conda-forge
referencing               0.32.0             pyhd8ed1ab_0    conda-forge
reportlab                 4.0.8           py312he37b823_0    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
rlpycairo                 0.2.0              pyhd8ed1ab_0    conda-forge
rpds-py                   0.13.2          py312h5280bc4_0    conda-forge
ruamel.yaml               0.18.5          py312he37b823_0    conda-forge
ruamel.yaml.clib          0.2.7           py312h02f2b3b_2    conda-forge
scipy                     1.11.4                   pypi_0    pypi
setuptools                68.2.2             pyhd8ed1ab_0    conda-forge
shadie                    0.2.0                     dev_0    <develop>
six                       1.16.0             pyh6c4a22f_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
svgwrite                  1.4.3              pyhd8ed1ab_0    conda-forge
tk                        8.6.13               h5083fa2_1    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
tornado                   6.3.3           py312h80987f9_0  
toyplot                   1.0.3              pyhd8ed1ab_0    conda-forge
toytree                   2.0.5              pyhd3deb0d_0    conda-forge
traitlets                 5.14.0             pyhd8ed1ab_0    conda-forge
tskit                     0.5.6           py312hf635c46_0    conda-forge
types-python-dateutil     2.8.19.14          pyhd8ed1ab_0    conda-forge
typing-extensions         4.9.0                hd8ed1ab_0    conda-forge
typing_extensions         4.9.0              pyha770c72_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
urllib3                   2.1.0              pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.12             pyhd8ed1ab_0    conda-forge
wheel                     0.42.0             pyhd8ed1ab_0    conda-forge
xorg-libxau               1.0.11               hb547adb_0    conda-forge
xorg-libxdmcp             1.1.3                h27ca646_0    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
zeromq                    4.3.4                hc377ac9_0  
zipp                      3.17.0             pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               h53f4e23_5    conda-forge
zstd                      1.5.5                h4f39d0f_0    conda-forge
```

**Comment by elissasoroj** [comment]
Hello Hannes,

I am glad you were able to figure out the install. I also have more luck installing packages separately. Like I said in my previous reply, I am in the middle of updating shadie and I would strongly suggest you don't use it for anything serious until I am done. I believe the current version of shadie you have installed uses SLiM3, but the new models have been optimized for SLiM4 and will run much more efficiently (and have also been checked for accuracy more recently than the SLiM3 models). I admit that no one but me has been using shadie, so I haven't been actively maintaining a working version on GitHub. 

I am aware of the issue with the toytree dependency. That package is undergoing a major upgrade and I have been waiting for the release to update the shadie dependencies in order to avoid asking users to install a dev version of toytree. If you really want to try shadie out you can try installing the [toy3-dev3](https://github.com/eaton-lab/toytree/tree/toy3-dev3) branch of toytree, but I can't guarantee that will fix the problem. 

Again, if you wait about a week or so shadie should be all updated, with updated tutorials and I can help you trouble-shoot any problems from there. 

All the best,
~Elissa

**Comment by hannesbecher** [comment]
Thanks for clarifying (again)!
Fee free to close this then.
Cheers,
Hannes

**Comment by elissasoroj** [comment]
Hi Hannes,
I just updated shadie for SLiM 4.1. You will also need the toy3 branch of toytree - there are updated instructions in the readme. 
I didn't get the chance to do extensive testing, but if you're not trying to do anything crazy it should work for you. I updated the first few tutorials, but I will need to go through and update the rest and also publish some of the ones that are kicking around in draft limbo. 
Let me know if you have any questions!

All the best,
~Elissa
